### PR TITLE
add v0.25 mkdocs, remove v0.21 hugo

### LINKED ISSUE #4117: Provide 0.25 docs version on knative.dev website
We need to add to the drop down `0.25` of the docs (mkdocs based) at the top.

And remove version `0.21`, this version is hugo based
image

After this change `0.25`, `0.24`, and `pre-release` will be in mkdocs format. The versions `0.23` and `0.22` will be in hugo format.

### PR DESCRIPTION
**Pull request by csantanapr**
Updated build script to add new version v0.25 mkdocs
Remove v0.21 hugo version

Closes #4117 

/assign @julz

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -19,10 +19,11 @@ set -x
 # 1) Make a release-NN branch as normal.
 # 2) Update VERSIONS below (on main) to include the new version.
 #    Order matters :-), Most recent first.
-VERSIONS=("0.24" "0.23" "0.22" "0.21")                  # Docs version, results in the url e.g. knative.dev/docs-0.23/..
-RELEASE_BRANCHES=("v0.24.0")                     # Release version for serving/eventing yaml files and api references.
+VERSIONS=("0.25" "0.24" "0.23" "0.22")                  # Docs version, results in the url e.g. knative.dev/docs-0.23/..
+VERSIONS_GENERATORS=("mkdocs" "mkdocs" "hugo" "hugo")  # update this to always be 4 in the next two releases replace hugo with mkdocs, remove the copy of static hugo site at the bottom
```
Comment (julz) [inline hack/build.sh:23]: Nice idea 👍

File: hack/build.sh [hunk 2]
```diff
@@ -19,10 +19,11 @@ set -x
...
+RELEASE_BRANCHES=("v0.25.0")                     # Release version for serving/eventing yaml files and api references.
```
Comment (julz) [inline hack/build.sh:24]:
```suggestion
RELEASE_BRANCHES=("v0.25.0" "v0.24.0")                     # Release version for serving/eventing yaml files and api references.
```

**Review by julz — COMMENTED** [review]
looks like the version warning banners got broken at some point, this is what I see in the header for 0.24 (note the lack of version number after the colon):

image

(don't think this PR broke it, but we should fix before merging)

Looks like the reference api downloads didn't quite work either, see https://deploy-preview-4151--knative.netlify.app/v0.24-docs/reference/api/serving-api/ - I think that's just because we need to update `RELEASE_BRANCHES=("v0.25.0") ` to `RELEASE_BRANCHES=("v0.25.0" "v0.24.0") `

File: hack/build.sh (hunk 2 above)
Comment (julz) [reply hack/build.sh:24]: I think this will fix both the lack of a version number in the banner and the lack of reference docs actually.

Comment (csantanapr) [reply hack/build.sh:24]: Good catch, thank you for the review. It looks like we are not catching an error when we download the API doc, adding `-L` to handle the redirect to a nonexisting endpoint.

Comment (csantanapr) [reply hack/build.sh:24]: ok verified now with `-L` we will catch the error like in this netlify build error
https://app.netlify.com/sites/knative/deploys/6123d10ead5dde00081754bf?utm_source=github&utm_campaign=bot_dl#L228-L240

**Comment by csantanapr** [comment]
I'm thinking on doing the removal of the archived 0.21 on another PR, this way this PR is cleaner to review the diff in the future.

**Comment by csantanapr** [comment]
@julz the problem is fixed now.

- **version shows up in warning banner**
image

- **API docs show up**
https://deploy-preview-4151--knative.netlify.app/v0.24-docs/reference/api/serving-api/
https://deploy-preview-4151--knative.netlify.app/v0.24-docs/reference/api/eventing-api/

- **Samples show correct branch**
image

I think it should be ready to merge

**Review by julz — APPROVED** [review]
Looks great! Thanks for doing the release @csantanapr 🎉 

/lgtm
/approve

**Pull request merged**